embassy-stm32/src/pwm/mod.rs

```diff
@@ -67,6 +70,14 @@ pub(crate) mod sealed {
         unsafe fn get_max_compare_value(&self) -> u16;
     }
 
+    pub trait ComplementaryCaptureCompare16bitInstance: CaptureCompare16bitInstance {
```

Comment:
Can you add these methods to `CaptureCompare16bitInstance` instead, and not add traits for `Complementary`?

The purpose of these traits is to keep track which timer supports which features. If all `CaptureCompare16bitInstance` timers also impl `ComplementaryCaptureCompare16bitInstance` then 1 trait is enough. The complementary driver can use the `CaptureCompare16bitInstance` trait.

Reply:
All `CaptureCompare16bitInstance` do not impl. `ComplementaryCaptureCompare16bitInstance`, only advanced timers do in most (all?) chips.

Reply:
oh, my bad! I misread the `foreach_interrupt!`. I see GP16/GP32 implement `CaptureCompare16bitInstance`, ADV impls both. Sorry!